app/src/main/java/com/lastaoutdoor/lasta/ui/screen/settings/SettingsComponent.kt

```diff
@@ -31,131 +43,218 @@ fun SettingsComponent(
+      // put a small setting icon in secondary color
+      Icon(
+          imageVector = Icons.Default.Build,
+          contentDescription = null,
+          tint = MaterialTheme.colorScheme.secondary,
+          modifier = Modifier.padding(8.dp))
+    } else {
+      // put a small setting icon in secondary color
+      Icon(
+          imageVector = Icons.Default.Settings,
+          contentDescription = null,
+          tint = MaterialTheme.colorScheme.secondary,
+          modifier = Modifier.padding(8.dp))
     }
   }
+}
 
-  Row {
-    Column(horizontalAlignment = Alignment.CenterHorizontally) {
-      // Outdoor activity selection
-      Text(
-          text = LocalContext.current.getString(R.string.select_fav_activity),
-          style = MaterialTheme.typography.headlineMedium,
-          color = MaterialTheme.colorScheme.onBackground)
+@Composable
+fun LanguageSelectionComponent(
+    language: Language,
+    updateLanguage: (Language) -> Unit,
+) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = LocalContext.current.getString(R.string.select_languague),
+        style = MaterialTheme.typography.headlineMedium,
+        color = MaterialTheme.colorScheme.onBackground)
+    Spacer(modifier = Modifier.height(8.dp))
+    Button(
+        onClick = {},
+        modifier = Modifier.testTag("languageDropDownButton"),
+        enabled = true,
+        shape = MaterialTheme.shapes.small,
+        colors =
+            ButtonDefaults.buttonColors(
+                containerColor = MaterialTheme.colorScheme.surface,
+                contentColor = MaterialTheme.colorScheme.onBackground,
+                disabledContainerColor = MaterialTheme.colorScheme.surfaceVariant,
+                disabledContentColor =
+                    MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.3f),
+            ),
+        elevation = ButtonDefaults.elevatedButtonElevation(1.dp)) {
+          DropDownMenuComponent(
+              items = Language.values().toList(),
+              selectedItem = language,
+              onItemSelected = { newLanguage: Language -> updateLanguage(newLanguage) },
+              toStr = { language -> language.resourcesToString(LocalContext.current) },
+              fieldText = LocalContext.current.getString(R.string.languague),
+              modifier = Modifier.testTag("settingsLanguage"))
+        }
+  }
+}
 
-      Spacer(modifier = Modifier.height(8.dp))
+@Composable
+fun FavoriteActivityComponent(
+    prefActivity: ActivityType,
+    updatePrefActivity: (ActivityType) -> Unit,
+) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = LocalContext.current.getString(R.string.select_fav_activity),
+        style = MaterialTheme.typography.headlineMedium,
+        color = MaterialTheme.colorScheme.onBackground)
+    Spacer(modifier = Modifier.height(8.dp))
+    Row(modifier = Modifier.testTag("settingsFavActivity")) {
```

Comment:
Here, it would be wise to use a FlowRow as it was done for the FilterScreen, otherwise the buttons will compensate with vertical space if they can't display all of their content horizontally
```suggestion
     FlowRow(
                verticalArrangement = Arrangement.spacedBy(4.dp),
                horizontalArrangement = Arrangement.SpaceEvenly,
                modifier = Modifier.padding(horizontal = 8.dp, vertical = 16.dp).fillMaxWidth()) {
                  activities.forEach { activity ->
                    ToggleButton(
                        activity.resourcesToString(LocalContext.current),
                        isSelected = selectedActivitiesTypes.contains(activity)) {
                          if (!selectedActivitiesTypes.contains(activity)) {
                            selectedActivitiesTypes.add(activity)
                          } else {
                            selectedActivitiesTypes.remove(activity)
                          }
                        }
                  }
                }
```

Reply:
yep good advice thanks !